app/middlewares/db_middleware.py

```diff
@@ -36,15 +37,19 @@ async def setup_chat(
         self, data: dict, user: types.User, chat: Optional[types.Chat] = None
     ):
         try:
+            chat_repo = ChatRepo()
```

Comment:
let's open here session (in_transaction context manager) and close it here in `finally` block

Reply:
I'm not sure if context manager will work here, since transaction will be closed when this middleware passes.
Do you mean to do something like this:
```suggestion
            session = in_transaction.__enter__()
            chat_repo = ChatRepo(session)
```
And `in_transaction.__exit__()` in the outer middleware?

Reply:
there are no pre and post middleware in aiogram 3.0

look at example (with sqlalchemy) at my template: https://github.com/bomzheg/aiogram_template/blob/master/app/middlewares/db_middleware.py